### DISCUSSION TITLE
Update on aub.ai: Navigating Challenges & Future Directions

### DISCUSSION THREAD

**Discussion by BrutalCoding** [BrutalCoding/aub.ai#16] [Announcements]
# Intro
Hi there,

First off - thanks. I had 2 goals with this project:
1. Bring gen-ai features to Flutter.
2. Secure myself a new source of income.

## Timeline from Sept 2023 - Feb 2024, this is why aub.ai was born.
Relevant context: I'm originally from The Netherlands (NL) (Holland), then moved to Australia for a holiday in 2019 and got stuck - long story short: I moved to a new country 25 hours away with the airplane. Aka, its far from my previous home in NL since 2019.

So, with that out of the way, let's fast forward to ...
1. Sept 2023: Company I worked for in Sydney (AU) went bankrupt. I lost my job. But that's the risk I was willing to take, part of the start-up life!
2. Oct 2023: My dad (in NL) passed away, fully healthy but an incident happened. Still unbelievable btw. Thus, I flew ~25 hours back to The Netherlands (NL) for this reason. Been 4 years since I saw my parents / fam.
3. Nov 2023: Doing family matters and I started prepping myself for job interviews, as I always do.
4. Dec 2023: My money is running out. I had been working on my on-device gen-ai Flutter plugin and decided to name it "aub.ai", together with open sourcing it. There was no other Flutter plugin that used llama.cpp in a workable state (+ up-to-date). I filled the gap there.
5. Dec '23/Jan '24: Flew back to Australia because I had no choice but to get back to my other fam there and start focusing on a new source of income. 
6. Jan 2024: Done 3 job interviews, ended up collabrating with someone that actually found me through aub.ai. Turns out that putting effort into aub.ai wasn't unnoticed.
7. Feb 2024: I got a new job now, and I am lucky to say that I am able to continue to expand my knowledge that aligns perfectly with what I was doing with aub.ai.

## My false promises
I did make some promises to users that support aub.ai, and I couldn't fulfill some of these, sorry for that. One such false promise of mine is this: https://github.com/BrutalCoding/aub.ai/issues/8#issuecomment-1905048528

## Where are the updates?
While I haven't contributed to aub.ai in a while, I did make progress. Lot of them actually. However, this was done under a difference license. I relicensed aub.ai to fit within a commercial project. This is the dual-license approach that I have stated in aub.ai's README since the start, but I did not expect this to happen so soon after release in Dec 2023.

## Future development
Look, I got a source of income, and am also continuing to expand my knowledge with on-device gen-ai within Flutter.

But as for future development, I need to figure out with the commercial party I work with what stuff I can contribute back open-source. Obviously, this is stuff that I do outside of work hours. But you can imagine that it would be a bit off-putting if I would release the same features that I had implemented in the commercial project.

Short answer: I need to bring this to topic with the commercial party I work with and come to a fair agreement.

There is more than on-device LLMs, I got a hang of more gen-ai features nowadays but I don't want to share details without having a chat with the commercial party I work with first. 

I hope you understand!

Stay tuned, I am grateful for all the support I have been receiving and am doing my best in my ability to start being active again with this project.

If you got any questions, go ahead and ask them here.

**Comment by KJ-Waller** [comment]
Hi Daniel, thank you for the comprehensive update. It's great to hear that you've secured a new job amidst the challenges. We all get how tough it can be to juggle a project like aub.ai by yourself. Honestly, the whole experience, whatever direction it takes, is super insightful and shows off what's possible with running LLMs on local devices. Big kudos to you for all the hard work. Can't wait to see what's next for you and aub.ai. Keep rocking it!
